From 7dddfb8fec5317ea16154d30e8e18b6559979b60 Mon Sep 17 00:00:00 2001
From: Bruce Ashfield <bruce.ashfield@windriver.com>
Date: Sun, 25 Aug 2013 22:51:07 -0400
Subject: [PATCH] ptrace.h: remove ptrace_peeksiginfo_args

The addition of ptrace_peeksiginfo_args to the uapi in kernel commit
84c751bd [ptrace: add ability to retrieve signals without removing from a queue (v4)]
means that existing applications using glibc versions that define ptrace_peeksiginfo_args
in sys/ptrace.h will get duplicate structure definitions like:

    | In file included from /poky-master/build/tmp/work/i586-poky-linux/strace/4.8-r0/strace-4.8/process.c:66:0:
    | /poky-master/build/tmp/sysroots/qemux86/usr/include/linux/ptrace.h:58:8: error: redefinition of 'struct ptrace_peeksiginfo_args'
    |  struct ptrace_peeksiginfo_args {
    |         ^
    | In file included from /poky-master/build/tmp/work/i586-poky-linux/strace/4.8-r0/strace-4.8/defs.h:159:0,
    |                  from /poky-master/build/tmp/work/i586-poky-linux/strace/4.8-r0/strace-4.8/process.c:37:
    | /poky-master/build/tmp/sysroots/qemux86/usr/include/sys/ptrace.h:191:8: note: originally defined here
    |  struct ptrace_peeksiginfo_args
    |         ^
    | make[2]: *** [process.o] Error 1

Reverting to the previous status of not exporting this structure temporarily
fixes applications, until they can be adjusted to not mix sys/ptrace.h and
linux/ptrace.h includes.

Signed-off-by: Bruce Ashfield <bruce.ashfield@windriver.com>
---
 include/uapi/linux/ptrace.h |    6 ------
 1 file changed, 6 deletions(-)

diff --git a/include/uapi/linux/ptrace.h b/include/uapi/linux/ptrace.h
index 52ebcc8..524599d 100644
--- a/include/uapi/linux/ptrace.h
+++ b/include/uapi/linux/ptrace.h
@@ -55,12 +55,6 @@
 
 #define PTRACE_PEEKSIGINFO	0x4209
 
-struct ptrace_peeksiginfo_args {
-	__u64 off;	/* from which siginfo to start */
-	__u32 flags;
-	__s32 nr;	/* how may siginfos to take */
-};
-
 /* Read signals from a shared (process wide) queue */
 #define PTRACE_PEEKSIGINFO_SHARED	(1 << 0)
 
-- 
1.7.10.4

